core/sail/nativerdf/src/main/java/org/eclipse/rdf4j/sail/nativerdf/TxnStatusFile.java

```diff
@@ -77,37 +82,48 @@ public void close() throws IOException {
 
 	/**
 	 * Writes the specified transaction status to file.
-	 * 
+	 *
 	 * @param txnStatus The transaction status to write.
 	 * @throws IOException If the transaction status could not be written to file.
 	 */
 	public void setTxnStatus(TxnStatus txnStatus) throws IOException {
-		byte[] bytes = txnStatus.name().getBytes(US_ASCII);
+		byte[] bytes = txnStatus.onDisk;
 		nioFile.truncate(bytes.length);
 		nioFile.writeBytes(bytes, 0);
 	}
 
 	/**
 	 * Reads the transaction status from file.
-	 * 
+	 *
 	 * @return The read transaction status, or {@link TxnStatus#UNKNOWN} when the file contains an unrecognized status
 	 *         string.
 	 * @throws IOException If the transaction status file could not be read.
 	 */
 	public TxnStatus getTxnStatus() throws IOException {
 		byte[] bytes = nioFile.readBytes(0, (int) nioFile.size());
-		String s = new String(bytes, US_ASCII);
-		try {
-			return TxnStatus.valueOf(s);
-		} catch (IllegalArgumentException e) {
-			// use platform encoding for backwards compatibility with versions
-			// older than 2.6.6:
-			s = new String(bytes);
-			try {
-				return TxnStatus.valueOf(s);
-			} catch (IllegalArgumentException e2) {
-				return TxnStatus.UNKNOWN;
+		TxnStatus status = TxnStatus.UNKNOWN;
+
+		if (bytes.length == 1) {
+			switch (bytes[0]) {
+			case 1:
+				status = TxnStatus.NONE;
+				break;
+			case 2:
+				status = TxnStatus.ACTIVE;
+				break;
+			case 3:
+				status = TxnStatus.COMMITTING;
+				break;
+			case 4:
+				status = TxnStatus.ROLLING_BACK;
+				break;
+			case 5:
+				status = TxnStatus.UNKNOWN;
+				break;
 			}
 		}
+
+		return status;
+
```

Comment:
I should add a try catch or maybe a default case that uses the old strings based approach.